src/Tribe/Views/V2/Views/Day_View.php

```diff
@@ -174,11 +236,25 @@ protected function build_url_for_date( $url_date, $canonical, array $passthru_va
 	 * {@inheritDoc}
 	 */
 	protected function setup_template_vars() {
+		$template_vars   = parent::setup_template_vars();
+		$sorted_events   = $this->sort_events( $template_vars['events'] );
+
+		$today           = Dates::build_date_object( $template_vars['url_event_date'] );
+		$next_date       = Dates::build_date_object( $template_vars['url_event_date'] )->add( new \DateInterval( 'P1D' ) )->format( 'Y-m-d' );
+		$prev_date       = Dates::build_date_object( $template_vars['url_event_date'] )->sub( new \DateInterval( 'P1D' ) )->format( 'Y-m-d' );
+
+		$next_event_date = $this->get_next_event_date( $today );
+		$prev_event_date = $this->get_previous_event_date( $today );
```

Comment:
Here is my fear this is incredibly expensive to run every time we build the template vars. It's two extra queries we have my and @borkweb removed back when we were running performance improvements to the Views V2.

Reply:
I get that there is a performance hit - but I don't see a way around the queries (although we can cache them) since we need to know _when_ the next/prev event is. Open to suggestions.

Reply:
I've added caching to the called functions to help with this (`get_next_event_date` & `get_previous_event_date`)